src/constructs/core/stack.ts

```diff
@@ -83,6 +83,13 @@ export class GuStack extends Stack implements StackStageIdentity, GuMigratingSta
    * Mappings to work around this limitation.
    */
   withStageDependentValue<T extends GuMappingValue>(mappingValue: GuStageMappingValue<T>): T {
+    const isInfraStageProvided = Object.keys(mappingValue.stageValues).includes(StageForInfrastructure);
+    if (isInfraStageProvided) {
+      Annotations.of(this).addWarning(
```

Comment:
It's worth noting that some of our constructs currently use (`withStageDependentValue`) under the hood - [ASGs are one example which come to mind](https://github.com/guardian/cdk/blob/d001f357814afc905c59c397dc60ae9a278cf200/src/constructs/autoscaling/asg.ts#L63-L86). That is, I think this warning will sometimes appear when users are not able to avoid it?

That said, I don't think this problem applies to any of the example `INFRA` services that you've listed, so perhaps we can revisit this in the future. (In theory it would apply to a service like Security HQ, which is currently `PROD`-only but might be better suited to the `INFRA` stage).

Reply:
Oops, you mentioned this on Chat too and I just didn't clock it, sorry!

Yes, this is a legit concern. As you said, it's currently unlikely that we'd be impacted on any of our current infrastructure stacks. I think we can make two improvements to this PR to help this:
- Add a TODO on the function you've linked to
- Add an annotation to the overridden `withStageDependentValue` function as a way to point users to consider using a vanilla variable rather than a CFN mapping

Reply:
Looking at it [more closely](https://github.com/guardian/cdk/search?q=withStageDependentValue) usage of `withStageDependentValue` is a bit wider than I had realised. I think we can make these usages support `INFRA` fairly easily, will give it a go.

Reply:
> Looking at it [more closely](https://github.com/guardian/cdk/search?q=withStageDependentValue) usage of `withStageDependentValue` is a bit wider than I had realised. I think we can make these usages support `INFRA` fairly easily, will give it a go.

Done on https://github.com/guardian/cdk/pull/948/commits/e684aa6c6c85f412880d401217c0a1448c908351. Existing tests continue to pass, which is enough, right?! 😅

Reply:
Thanks for revisiting this, the latest changes look good to me 👍

Reply:
> Existing tests continue to pass, which is enough, right?! 😅

Generally our test coverage is pretty good, so let's hope so!